staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -23,12 +23,12 @@ import (
 )
 
 // ValidateListOptions returns all validation errors found while validating the ListOptions.
-func ValidateListOptions(options *internalversion.ListOptions) field.ErrorList {
+func ValidateListOptions(options *internalversion.ListOptions, isWatchListFeatureEnabled bool) field.ErrorList {
+	if options.Watch {
+		return validateWatchOptions(options, isWatchListFeatureEnabled)
+	}
```

Comment:
There are some validations below (e.g. Continue or the fact that ResourceVersionMatch is not a random string) that we should share chagin between list  and watch.

Reply:
we do check on `ResourceVersionMatch`, `continue` param was missing (added). 

In general I don't think this validation is complete, for example you could start a watch or a list with `continue` param set which doesn't make sense.